src/main/java/io/neonbee/endpoint/odatav4/ODataV4Endpoint.java

```diff
@@ -220,19 +238,38 @@ public Router createEndpointRouter(Vertx vertx, String basePath, JsonObject conf
             // before adding new routes, get a list of existing routes, to remove after the new routes have been added
             List<Route> existingRoutes = router.getRoutes();
 
-            // Register new routes first, this will avoid downtimes of already existing services. Register the shortest
+            // register new routes first, this will avoid downtimes of already existing services. Register the shortest
             // routes last, this will lead to some routes like the empty namespace / to be registered last.
             models.values().stream().flatMap(entityModel -> entityModel.getEdmxes().entrySet().stream())
                     .map(entryFunction(
                             (schemaNamespace, edmxModel) -> Map.entry(uriConversion.apply(schemaNamespace), edmxModel)))
                     .sorted(Map.Entry.comparingByKey(Comparator.comparingInt(String::length).reversed()))
                     .forEach(entryConsumer((uriPath, edmxModel) -> {
-                        // TODO depending on the config either create Olingo or CDS based OData V4 handlers here
+                        String schemaNamespace = edmxModel.getEdm().getEntityContainer().getNamespace();
                         router.route((uriPath.isEmpty() ? EMPTY : ("/" + uriPath)) + "/*")
+                                // some entities should not get exposed, register a handler, checking the block list
+                                .handler(routingContext -> {
+                                    // normalize the URI first
+                                    NormalizedUri normalizedUri = normalizeUri(routingContext, schemaNamespace);
+                                    if (LOGGER.isDebugEnabled()) {
+                                        LOGGER.correlateWith(routingContext).debug("Normalized OData V4 URI {}",
+                                                normalizedUri);
+                                    }
+
+                                    // if a entity is specified check it against the block list
+                                    // TODO: maybe also navigation properties have to be taken into account here?
+                                    if (normalizedUri.fullQualifiedName != null
```

Comment:
I'm not sure how expensive this regexp check is, but it will be performed for every request. The allow/block list cannot be changed after startup, maybe we should store the regexp results in a Map.